src/main/java/uk/ac/ebi/ampt2d/registry/entities/EntityEventListener.java

```diff
@@ -0,0 +1,54 @@
+/*
+ *
+ * Copyright 2018 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.registry.entities;
+
+import uk.ac.ebi.ampt2d.registry.util.BeanUtil;
+import uk.ac.ebi.ampt2d.registry.service.mail.MailService;
+
+import javax.persistence.PostPersist;
+import javax.persistence.PostRemove;
+import javax.persistence.PostUpdate;
+
+public class EntityEventListener {
+
+    private enum Event {
+        CREATED,
+        UPDATED,
+        REMOVED
+    }
+
+    @PostPersist
```

Comment:
You can use @HandleBeforeCreate/@HandleBeforeSave as this more with spring; your choice

Reply:
We try to favour standard annotations over Spring ones.